californium-core/src/main/java/org/eclipse/californium/core/network/InMemoryMessageExchangeStore.java

```diff
@@ -353,12 +325,8 @@ public boolean registerOutboundResponse(final Exchange exchange) {
 	 * Purges all registered exchanges from this store.
 	 */
 	public void clear() {
-		synchronized (messageIdProvider) {
-			synchronized (exchangesByToken) {
-				exchangesByMID.clear();
-				exchangesByToken.clear();
-			}
-		}
+		exchangesByMID.clear();
```

Comment:
OK, T1 is adding to `exchangesByMID` and then to `exchangesByToken`.
And in between T2 invokes `clear()`.
My understand was, that `clear()` is only useful on shutdown and so I don't see that this will end in too much additional problems. Or is there any other use-case to invoke `clear()`?